### ISSUE TITLE
Pool size increase: MaxDoorExtensions 

### ISSUE THREAD

**Issue by kamelzarandah** [citizenfx/fivem#2888]
### What happened?

MaxDoorExtensions value="320"
if possible can we also get this pool size, as of right now, exceeding this one also causes an instant crash
RDR2_b1491.exe!?Entities_Create@fwMapDataContents@rage@@s?_PPAVfwMapData@rage@@PAVfwMapDataContents@Entities_Create@@h@Z (0x10a)

Door extensions crash (if you have to many Door extensions in ymap or ytyp Files)

something like this happens when you have many big apartment buildings for example in Saint Denis where all the big buildings are next to each other

### Expected result

ability to add more doors without crashing 

### Reproduction steps

a few maps with doors next to each other will result in a pool size crash related to MaxDoorExtensions 

### Importancy

Crash

### Area(s)

RedM, FXServer

### Specific version(s)

redm/ windows 10573

### Additional information

increasing this pool size can help with the task of opening up the whole map with mlo's and ymaps

**Comment by LandminenTester** [comment]
I support this!

**Comment by SevenLife** [comment]
I support it too ! Like custom/edited AMV pool size limit ! they need to be raised please

**Comment by farsey87** [comment]
yes please

**Comment by Masterspooni** [comment]
I support this!

**Comment by iSn4x** [comment]
Yes, pls let us pick the pool sizes we need

**Comment by M8-4-Life** [comment]
Need this please, definitely support this!

**Comment by MissGhouls** [comment]
Definitely support this! We desperately need this.

**Comment by BlackLegion23** [comment]
Absolutly supporting this

**Comment by Perla39** [comment]
I support this!

**Comment by outsider31000** [comment]
@kamelzarandah  according to the list https://content.cfx.re/mirrors/client/pool-size-limits/redm.json this wasn't added yet, but I think its possible to increment using the new features to increase the pool sizes, have you tried ?

@Nobelium-cfx  is it possible to add this to the list? or it is not required anymore?
other wise should be good to close this issue.

**Comment by Nobelium-cfx** [comment]
> [@kamelzarandah](https://github.com/kamelzarandah) according to the list https://content.cfx.re/mirrors/client/pool-size-limits/redm.json this wasn't added yet, but I think its possible to increment using the new features to increase the pool sizes, have you tried ?
> 
> [@Nobelium-cfx](https://github.com/Nobelium-cfx) is it possible to add this to the list? or it is not required anymore? other wise should be good to close this issue.

From a quick look to it - it seems that `MaxDoorExtensions` pool is not natively supported by [increase-pool-size](https://docs.fivem.net/docs/server-manual/server-commands/#increase_pool_size-poolname-increase) command. It uses the same mechanism as [Pool Monitor](https://github.com/citizenfx/fivem/blob/master/code/components/gta-core-rdr3/src/PoolManagement.cpp) and I don't see `MaxDoorExtensions` there.

But again - it's just a quick look and I may be wrong. You can use the local experiments feature (https://docs.fivem.net/docs/server-manual/server-commands/#local-experiments), try to increase it locally and see if it actually works.

If it does - let me know, we will add it to the allow list then.

If not - we would either need to take a closer look at how to add support for it, or just try to increase it in https://github.com/citizenfx/fivem/blob/master/data/client_rdr/citizen/common/data/gameconfig.xml . Second is probably more pragmatic approach. So then feel free to make a PR increasing it in (linking this discussion).

**Comment by Identity-labs** [comment]
All pools are available in pool monitor, but not all pools have name hash.
Check if jooat of MaxDoorExtensions is present, if not MaxDoorExtensions is not a valid pool name

**Comment by outsider31000** [comment]
> All pools are available in pool monitor, but not all pools have name hash. Check if jooat of MaxDoorExtensions is present, if not MaxDoorExtensions is not a valid pool name

CPed and CObject also do not show there theres only CVehicle

**Comment by Identity-labs** [comment]
because it's Peds and Objects

**Comment by outsider31000** [comment]
> because it's Peds and Objects

but why using  GetGamePool("CPed") works ? and Peds dont ?

![Image](https://github.com/user-attachments/assets/c922d909-0afb-4b67-8fad-b8925ef4f482)

thats what I am confused same happens for CObject

**Comment by Identity-labs** [comment]
Because it's hardcoded in fivem code
https://github.com/citizenfx/fivem/blob/master/code/components/extra-natives-five/src/PoolTraversalNatives.cpp#L289

**Comment by outsider31000** [comment]
its confusing then using CPeds when its called Peds in the Pool monitor

But they still dont show on the monitor Pool those 2 that I mentioned, only Peds show not even Objects show

**Comment by outsider31000** [comment]
ah its `Object`

**Comment by outsider31000** [comment]
yeah this name `MaxDoorExtensions` doesn't seem to be a valid pool name 

@Nobelium-cfx  the hash though is most likely this one `0xb4e04aa4` since is the only pool with max items 320

**Comment by Gogsi** [comment]
Yes it is that one. The name of the pool is CDoorExtension.

![Image](https://github.com/user-attachments/assets/83248382-a7e5-4bda-9b30-50677278889e)

MaxDoorExtension is just the name of the field that the parser reads from CGameConfig to get the vanilla value for the CDoorExtension pool size. https://alexguirre.github.io/rage-parser-dumps/dump.html?build=2824&game=gta5#CExtensionConfig

**Comment by outsider31000** [comment]
oh nice, yeah its missing on the list ill adjust my pr then thanks

**Comment by Gogsi** [comment]
All of this also applies to V where the pool only has a size of 64 btw

**Comment by Nobelium-cfx** [comment]
> not sure how to test if it increases or not since this doesnt exist in the pool monitor to visually see the max items
unless there's a way to get the max items that I'm not aware of?

I was thinking about adding it and then checking if crashes reported in this bug reappear. But trying to add the name to the `poolEntriesTable` is also ok.

> Yes it is that one. The name of the pool is CDoorExtension.

Oh, thank you @Gogsi ! I checked Joaat hash - but only for `MaxDoorExtension` (couldn't find and decided it's just allocated differently), didn't think of trying other `CDoorExtension`. Hash `0xb4e04aa4` is indeed in the list.

Ok, so then we can add `CDoorExtension` to `poolEntriesTable` list and the allow increasing `CDoorExtension` through [increase-pool-size](https://docs.fivem.net/docs/server-manual/server-commands/#increase_pool_size-poolname-increase). I'll make a commit and reply here once it gets merged.

**Comment by outsider31000** [comment]
https://github.com/citizenfx/fivem/pull/3142
The pr is here with the correct name btw @Nobelium-cfx 
And thank you.

**Comment by Nobelium-cfx** [comment]
`CDoorExtension` should be added to Canary soon with https://github.com/citizenfx/fivem/commit/8871b05733a7533bb1db2e82fa56d62b8489d07b . Also updated https://content.cfx.re/mirrors/client/pool-size-limits/redm.json to allow increasing CDoorExtension by up to 160.

So it should be all done now. Feel free to reopen the issue if I missed something.

**Comment by kamelzarandah** [comment]
> `CDoorExtension` should be added to Canary soon with [8871b05](https://github.com/citizenfx/fivem/commit/8871b05733a7533bb1db2e82fa56d62b8489d07b) . Also updated https://content.cfx.re/mirrors/client/pool-size-limits/redm.json to allow increasing CDoorExtension by up to 160.
> 
> So it should be all done now. Feel free to reopen the issue if I missed something.

is it possible to take a look at CPropSetObjectExtension pool size, this would be the last one causing crashes for us at least. attached is the image of the crash when this pool size is full. its a very common crash for us. 

thank you so much for the effort, you and everyone who contributed to this thread! you guys make a huge difference everyday 

![Image](https://github.com/user-attachments/assets/46ec3354-084b-4f86-aa97-55a7fd243933)

**Comment by Nobelium-cfx** [comment]
@kamelzarandah Looks good. Done.

PS: It may take a few hours for the change to propagate.

**Comment by kamelzarandah** [comment]
what would be the new max size ?

**Comment by Nobelium-cfx** [comment]
> what would be the new max size ?

Currently you can increase it by 450. So 1400 in total.
